jupyter_book_to_htmlbook/code_processing.py

```diff
@@ -111,3 +112,67 @@ def number_codeblock(pre_block, cell_number):
                 element.replace_with(element.replace('\n', f'\n{indent}'))
 
     return cell_number
+
+
+def process_code_examples(chapter):
+    """
+    Applies appropriate data types and adds titles for formal code
+    "Examples" in the text
+    """
+    examples = chapter.find_all("div", class_="tag_example")
+
+    for example_cell in examples:
+        pre_block = example_cell.find("pre")
+        example_div = pre_block.parent
+        try:
+            comments = pre_block.find_all("span", class_="c1")
+
+            # ensure comments are within the first three spans (since we
+            # expect an empty span to start based on their highlighter)
+            if not (
+                    comments[0] in pre_block.find_all("span")[0:3] and
+                    comments[1] in pre_block.find_all("span")[0:3]
+                   ):
+                logging.warning(
+                    "Missing first two line comments for uuid and title." +
+                    f"Unable to apply example formatting to {example_cell}.")
+                return example_cell
+
+            # first comment should be uuid/name; only take first word if many
+            # to preserve id attribute validation
+            example_name = comments[0].string
+            example_name = example_name.split(" ")[1]  # remove comment hash
+            comments[0].decompose()
+            # second comment should be the example title
+            example_title = comments[1].string
+            example_title = " ".join(example_title.split(" ")[1:])
+            comments[1].decompose()
+
+            logging.info("Applying example formatting to and removing" +
+                         f" first comments from: {example_cell}")
+        except IndexError:
+            logging.warning(
+                    f"Unable to apply example formatting to {example_cell}")
+            return example_cell
+
+        # remove empty space left by decomposed spans
+        # assuming they add the empty span at the beginning of block, do [0:3]
+        for element in pre_block.contents[0:3]:
+            if (
+                    type(element) == NavigableString and
+                    element.string in ["\n", "\n\n"]
+               ):
+                element.replace_with('')
+
+        # apply data-type to immediate parent
```

Comment:
Idea: Another possible way to break up this method would be to have one function return the `example_name` and `example_title`, then have separate functions to apply them back onto the parent `<div>`. I don't know if that's worth it but just a thought.

Reply:
Oh, I like this idea!

Reply:
See updated code